server/src/test/java/org/elasticsearch/action/admin/indices/rollover/RolloverResponseTests.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.admin.indices.rollover;
+
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.XContentTestUtils;
+import org.elasticsearch.test.hamcrest.ElasticsearchAssertions;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+
+public class RolloverResponseTests extends ESTestCase {
```

Comment:
Should we also add serialization roundtrip tests to this new test class?

Reply:
yes it would be nice but I think it's besides the scope of this PR. I can add it but I must admit we didn't do it before for other responses and adding that may require adding equals/hashcode methods plus testing them etc. which becomes too time-consuming. There are enough tests added around the high-level REST client stuff already I think, and the main goal here is adding support for new APIs to the high-level REST client. If some important test is missing we should open separate issues for them and address them separately. What do you think?

Reply:
Np, was also more of a "nice to have" since we're introducing this new test here. Would you open a separate issue to improbe this at some point though?

Reply:
I couldn't help it, I didn't want to leave yet another issue behind, not knowing who and when would take care of it. I may go back and also look at some other response tests, given that implement a different base test class gives us almost all we need already.